src/client/common/process/pythonExecutionFactory.ts

```
@@ -2,23 +2,33 @@
-import { ExecutionFactoryCreationOptions, IProcessServiceFactory, IPythonExecutionFactory, IPythonExecutionService } from './types';
+import { ExecutionFactoryCreationOptions, IBufferDecoder, IProcessServiceFactory, IPythonExecutionFactory, IPythonExecutionService } from './types';
 
 @injectable()
 export class PythonExecutionFactory implements IPythonExecutionFactory {
-    private readonly configService: IConfigurationService;
-    private processServiceFactory: IProcessServiceFactory;
-    constructor(@inject(IServiceContainer) private serviceContainer: IServiceContainer) {
-        this.processServiceFactory = serviceContainer.get<IProcessServiceFactory>(IProcessServiceFactory);
-        this.configService = serviceContainer.get<IConfigurationService>(IConfigurationService);
+    constructor(@inject(IServiceContainer) private serviceContainer: IServiceContainer,
+        @inject(IEnvironmentActivationService) private readonly activationHelper: IEnvironmentActivationService,
+        @inject(IProcessServiceFactory) private readonly processServiceFactory: IProcessServiceFactory,
+        @inject(IConfigurationService) private readonly configService: IConfigurationService,
+        @inject(IBufferDecoder) private readonly decoder: IBufferDecoder) {
     }
     public async create(options: ExecutionFactoryCreationOptions): Promise<IPythonExecutionService> {
         const pythonPath = options.pythonPath ? options.pythonPath : this.configService.getSettings(options.resource).pythonPath;
         const processService = await this.processServiceFactory.create(options.resource);
         return new PythonExecutionService(this.serviceContainer, processService, pythonPath);
     }
+    public async createActivatedEnvironment(resource: Resource): Promise<IPythonExecutionService> {
+        const envVars = await this.activationHelper.getActivatedEnvironmentVariables(resource);
+        if (!envVars || Object.keys(envVars).length === 0) {
```

Does this basically swallow a bunch of possible errors silently? Looks like getActivatedEnvironmentVariables returns back undefined for a few cases and this will just create a standard ExecutionService without notifying us at all. If that's expected it's fine, but it looks like some of those cases might be issues versus expected.